R/plot_heatmap.R

```diff
@@ -0,0 +1,105 @@
+#' a mutational discovery or prediction tool.
+#' @param proportional If \code{TRUE}, then the exposures will be normalized
+#' to between 0 and 1 by dividing by the total number of counts for each sample.
+#' Default \code{FALSE}.
+#' @param annotations Column annotations extracted from the musicatk object 
+#' @param show_column_names Boolean check. If \code{True}, column names are shown. Otherwise, they aren't. 
+#' Default \code{FALSE}
+#' @param show_row_names Boolean check. If \code{True}, row names are shown. Otherwise, they aren't. 
+#' Default \code{FALSE}
+#' @param scale Boolean check. If \code{True}, values are scaled by z-score. Otherwise, they aren't.
+#' Default \code{TRUE}
+#' @param annotation Users have the option of plotting the exposure matrix based on their given annotation like
+#' Tumor_Subtypes or age. Error given if the user given annotation doesn't exist in the result annotation object.
+#' @param subset_tumor Users can specify certain tumor types on which they want to subset the exposure matrix for 
+#' plotting the heatmap. 
+#' @param subset_signatures Users can specify certain signatures on which they want to subset the exposure matrix 
+#' plotting the heatmap.
+#' @return Generates a heatmap for using the exposure matrix.
+#' @examples result <- readRDS(system.file("testdata", "res_annot.rds",
+#' package = "musicatk"))
+#' exp <- exposures(result)
+#' ComplexHeatmap::Heatmap(exp)
+#' @export
+
+
+
+plot_heatmap <- function(result, 
+                         proportional = FALSE,
+                         show_column_names = FALSE,
+                         show_row_names = TRUE,
+                         scale = TRUE,
+                         subset_tumor = NULL,
+                         subset_signatures = NULL,
+                         annotation = NULL,
+                         ...
+                         ){
+  
+  exp <- exposures(result) #extracting the exposures matrix
+  
+  if(isTRUE(proportional)){
+    exp <- sweep(exp, 2, colSums(exp), FUN = "/") #normalizing 
+  }
+  
+  if(isTRUE(scale)){
+    exp <- scale(exp)
+  }
+  
+  annot <- sample_annotations(result) #Extracting annotations from musicatk object
+  heatmap <- NULL #initializing an empty heatmap annotation variable
+  
+  #checking if annotation argument correct
+  if(!is.null(annotation)){
+    if(any(!annotation %in% colnames(annot))) { 
+      stop("The given annotations are not present in the data") 
+    }
+    annot <- annot[,..annotation]
+    heatmap <- ComplexHeatmap::HeatmapAnnotation(df = annot)
+  }
+  
+  if(!is.null(subset_signatures)){
+    if(any(!subset_signatures %in% rownames(exp))) { 
+      stop("The given signatures are not present in the data") 
+    }
+    exp <- as.data.frame(exp)
+    exp <- subset(exp,rownames(exp) %in% subset_signatures)
+    exp <- as.matrix(exp)
+  }
+  
+  if(!is.null(subset_tumor)){
+    annot <- sample_annotations(result)
+    samps <-  subset(annot, Tumor_Subtypes == subset_tumor)
```

Comment:
This needs to take in any possible annotation, "Tumor_Subtypes" is just an example value that an annotation could be. It could just as easily be "Sequencing_Center", "Batch", "Age", etc.